pkg/kubelet/kubelet.go

```diff
@@ -745,7 +746,26 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	// check node capabilities since the mount path is not the default
 	if len(kubeCfg.ExperimentalMounterPath) != 0 {
 		kubeCfg.ExperimentalCheckNodeCapabilitiesBeforeMount = false
+		resolvePath := filepath.Join(strings.TrimSuffix(kubeCfg.ExperimentalMounterPath, "/mounter"), "rootfs", "etc", "resolv.conf")
```

Comment:
Are there any issues with adding the dns servers to the host resolv.conf?  Wouldn't that also solve the hostname problem for non-containerized mounts?

Reply:
Adding kube-dns's service VIP to resolv.conf might introduce too many side effects:
- kube-dns's service VIP may not be resolvable on host in some environments (ref minkube: https://github.com/kubernetes/community/pull/833#issuecomment-320808512)
- Processes on host namespace will send DNS requests to kube-dns first, then kube-dns will forward requests to upstream. This brings in additional hop and latency.
- kube-dns pod copies resolv.conf from host in order to mirror the upstream DNS server setup, if we set kube-dns's service VIP there, this seems to be a circle?
- If kube-dns dies, it breaks the DNS resolution on host even if upstream server is healthy.

Reply:
And I think it will breaks `dnsPolicy` field on pod.spec as well :)

Reply:
I guess when we move to CSI, the volume mounts will happen inside a container, and no longer from the host, so in the meantime, for non-containerized mounts using containerized filesystem services, we still only support using IPs and not hostnames.